server/src/main/java/io/spine/server/delivery/Delivery.java

```diff
@@ -436,6 +441,27 @@ public Optional<DeliveryStats> deliverMessagesFrom(ShardIndex index) {
         return Optional.of(stats);
     }
 
+    /**
+     * Notifies the {@code DeliveryMonitor} about the technical failure and executes
+     * the failure handler {@code Action}.
+     */
+    private Optional<DeliveryStats> handleTechPickUpFailure(ShardIndex shard, RuntimeException e) {
+        TechFailure failure = new TechFailure(shard, e, () -> deliverMessagesFrom(shard));
+        return monitor.onShardPickUpFailure(failure)
+                      .execute();
+    }
+
+    /**
+     * Notifies the {@code DeliveryMonitor} that shard is already pocked up and executes
+     * the failure handler {@code Action}.
+     */
+    private Optional<DeliveryStats> handleAlreadyPickedUp(ShardIndex shard, WorkerId owner) {
+        AlreadyPickedUp failure =
+                new AlreadyPickedUp(shard, owner, () -> deliverMessagesFrom(shard));
+        return monitor.onShardAlreadyPicked(failure)
```

Comment:
Please extract the result into variable prior to returning.

It's a common practice to do so, especially with the chained calls like this one. Otherwise, it's really hard to inspect any issues in scope of debugging.